bundles/org.openhab.core.io.transport.upnp/src/main/java/org/openhab/core/io/transport/upnp/internal/UpnpIOServiceImpl.java

```diff
@@ -154,11 +155,11 @@ protected void eventReceived(GENASubscription sub) {
                     deviceRoot.getIdentity().getUdn());
             for (UpnpIOParticipant participant : participants) {
                 if (Objects.equals(getDevice(participant), deviceRoot)) {
-                    for (String stateVariable : values.keySet()) {
-                        StateVariableValue value = values.get(stateVariable);
+                    for (Entry<String, StateVariableValue> entry : values.entrySet()) {
+                        StateVariableValue value = entry.getValue();
                         if (value.getValue() != null) {
```

Comment:
I think this null-check can be omitted if we use `entry.getValue()` because the map has no null values.

Reply:
I'm getting NPE's I believe due to the removal of the null check here.  I've opened #4037 to track that.